pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023. SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.media3.common.Player
+import ch.srgssr.pillarbox.player.availableCommandsAsFlow
+import ch.srgssr.pillarbox.player.canSeek
+import ch.srgssr.pillarbox.player.currentPositionAsFlow
+import ch.srgssr.pillarbox.player.currentPositionPercent
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+
+/**
+ * Progress tracker
+ *
+ * Handle a progress position that is a mix of the player current position and the user seeked position.
+ *
+ * @property player The player to track current position.
+ */
+@Stable
+class ProgressTracker internal constructor(private val player: Player) {
+    private val playerProgressPercent: Flow<Float> = player.currentPositionAsFlow().map { player.currentPositionPercent() }
+    private val userSeekState = MutableStateFlow<UserSeekState>(UserSeekState.Idle)
+    private val canSeek = player.availableCommandsAsFlow().map { it.canSeek() }
+    private val progressPercentFlow: Flow<Float> = combine(userSeekState, playerProgressPercent) { seekState, playerProgress ->
+        when (seekState) {
+            is UserSeekState.Seeking -> seekState.percent
+            else -> playerProgress
+        }
+    }
+
+    /**
+     * Progress percent
+     *
+     * @return progress percent as State.
+     */
+    @Composable
+    fun progressPercent(): State<Float> = progressPercentFlow.collectAsState(initial = player.currentPositionPercent())
+
+    /**
+     * Can seek
+     *
+     * @return can seek as State.
+     */
+    @Composable
+    fun canSeek(): State<Boolean> = canSeek.collectAsState(initial = player.availableCommands.canSeek())
+
+    /**
+     * User seek at percent position
+     *
+     * @param percent Position in percent [0,1].
+     */
+    fun userSeek(percent: Float) {
+        userSeekState.value = UserSeekState.Seeking(percent)
+    }
+
+    /**
+     * User has finish seeking.
+     */
+    fun userSeekFinish() {
```

Comment:
```suggestion
    fun userSeekFinished() {
```

or

```suggestion
    fun finishUserSeek() {
```

maybe?

Reply:
I have a preference for `userSeekFinished`.